### PR TITLE
Fix for bug when using InterpND to interpolate on 1D data.

### LINKED ISSUE #1936: Exception when using Interp1D with a 1D table where the grid is defined with an ndarray.
### Summary of Issue

Exception when using Interp1D with a 1D table where the grid is defined with an ndarray instead of a list with a single ndarray in it.

### Issue Type

- [x ] Bug
- [ ] Enhancement  
      POEM #___ <N/A or POEM number at https://github.com/OpenMDAO/POEMs>

### Description

Error checking is really not being done in the proper order here. Points should probably be check before x-interp.

In addition, the docstrings seem very unclear and need an overhaul to concisely explain what the user should input.

### Example
https://stackoverflow.com/questions/66432593/interpnd-for-one-dimension/66440312#66440312

### Environment

<For bugs, otherwise N/A>

Operating System: <i.e. OS X 10.14.6, Windows 10, Ubuntu 16.04>

Python environment: <i.e. Anaconda Python 3.7.1>

Packages: <versions of Dymos, OpenMDAO, and other pertinent Python packages. You can just paste the output of `pip freeze` or `conda list` here.>

### PR DESCRIPTION
**Pull request by Kenneth-T-Moore**
### Summary

Fix for bug when using InterpND to interpolate on 1D data.  It now works when you 1-dimensional arrays as any input. Formerly those arrays needed to be 2D.

### Related Issues

- Resolves #1936

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: openmdao/components/interp_util/interp.py [hunk 1]
```diff
@@ -80,17 +80,30 @@ class InterpND(object):
     def __init__(self, method="slinear", points=None, values=None, x_interp=None, extrapolate=False,
                  num_cp=None, **kwargs):
         """
-        Initialize instance of interpolation class.
+        Initialize an InterpND object.
+
+        This object can be setup and used to interpolate on a curve or multi-dimensional table
```
Comment (naylor-b) [inline openmdao/components/interp_util/interp.py:85]: missing a period

File: openmdao/components/interp_util/interp.py [hunk 2]
```diff
@@ -80,17 +80,30 @@ class InterpND(object):
...
+
+        It can also be used to setup an interpolating spline that can be evaluated at fixed
+        locations.
+
+        For interpolation, specify values and points.
+
+        For spline evaulation, specifiy x_interp and either points or num_cp.
```
Comment (naylor-b) [inline openmdao/components/interp_util/interp.py:92]: evaulation

File: openmdao/components/interp_util/interp.py [hunk 3]
```diff
@@ -80,17 +80,30 @@ class InterpND(object):
...
-        values : array_like, shape (m1, ..., mn, ...)
-            The data on the regular grid in n dimensions.
         method : str
-            Name of interpolation method(s).
+            Name of interpolation method.
+        points : ndarray or tuple of ndarray
+            The points defining the regular grid in n dimensions.
+            For 1D interpolation, this can be an ndarray of table locations.
+            For table interpolation, it can be a tuple or an ndarray. If it is a tuple, it should
+            contain one ndarry for each table dimension.
```
Comment (naylor-b) [inline openmdao/components/interp_util/interp.py:102]: ndarry

File: openmdao/components/interp_util/interp.py [hunk 4]
```diff
@@ -113,8 +126,35 @@ def __init__(self, method="slinear", points=None, values=None, x_interp=None, ex
                              '%s.' % (method, all_m))
         self.extrapolate = extrapolate
 
+        # The table points are always defined, either by specifying the number of the points
+        # directly.
```
Comment (naylor-b) [inline openmdao/components/interp_util/interp.py:130]: This sentence has an 'either' but is it missing an 'or'?

**Review by naylor-b — CHANGES_REQUESTED** [review]

**Comment by Kenneth-T-Moore** [comment]
All issues fixed.

**Review by naylor-b — APPROVED** [review]

**Review by swryan — APPROVED** [review]

**Pull request merged**